### PR TITLE
Deprecate `prefect kubernetes manifest` command

### PR DESCRIPTION
**Pull request by enriched-h**
### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

File: src/prefect/cli/kubernetes.py [hunk 1]
```diff
@@ -24,9 +24,14 @@
...
 app.add_typer(kubernetes_app)
 
+# Deprecate kubernetes maniest
 manifest_app = PrefectTyper(
     name="manifest",
-    help="Commands for generating Kubernetes manifests.",
+    help="Deprecated. Use the 'Helm charts' instead.",
+    deprecated=True,
+    deprecated_name="prefect kubernetes manifest",
+    deprecated_start_date="4th May 2023",
```
Comment (madkinsz) [inline src/prefect/cli/kubernetes.py:33]: I believe this expects month and year but not day

File: src/prefect/cli/kubernetes.py [hunk 2]
```diff
@@ -184,3 +189,6 @@ async def manifest_flow_run_job():
     )
 
     print(output)
+"""
+
+"""
```
Comment (madkinsz) [inline src/prefect/cli/kubernetes.py:194]: Looks unintentional

File: src/prefect/cli/kubernetes.py [hunk 3]
```diff
@@ -24,9 +24,14 @@
 )
 app.add_typer(kubernetes_app)
 
+# Deprecate kubernetes maniest
 manifest_app = PrefectTyper(
     name="manifest",
-    help="Commands for generating Kubernetes manifests.",
+    help="Deprecated. Use the 'Helm charts' instead.",
```
Comment (madkinsz) [inline src/prefect/cli/kubernetes.py:30]: I think that we can retain the previous help as this should be covered by the `deprecated_help`. Maybe 
```suggestion
    help="Commands for generating Kubernetes manifests. Deprecated.",
```

File: src/prefect/cli/kubernetes.py [hunk 4]
```diff
@@ -24,9 +24,14 @@
...
+    deprecated=True,
+    deprecated_name="prefect kubernetes manifest",
+    deprecated_start_date="4th May 2023",
+    deprecated_help="Use the 'Helm charts' instead.",
```
Comment (madkinsz) [inline src/prefect/cli/kubernetes.py:34]: I think we can give a better suggestion here — maybe a link to the Helm charts?

cc @jawnsy

**Comment by madkinsz** [comment]
Can you update the body of the PR to:
- Describe the change
- Link to the issue this closes
- Include an example of running the command and the deprecation warning

**Comment by chrisguidry** [comment]
Thank you for your contribution, @enriched-h, we have two PRs open for the same issue, so we're going to unify around #9479.

**Pull request closed without merging**